### PR TITLE
CarPlayManager support Xcode 12

### PR DESCRIPTION
**Pull request by MarcoCarnevali**
small update to CarPlayManager.swift in order to support the new Xcode 12

### CODE REVIEW

**Comment by MarcoCarnevali** [comment]
@1ec5 Sorry, there was a problem with the commit. I just sent a new commit that should work. 
Thank you!

**Comment by aodhol** [comment]
@1ec5 any chance of geting this merged :)?

**Comment by Udumft** [comment]
Hey, @MarcoCarnevali! Thank you for submission!
I was trying to build your branch using XCode 12 Beta3 but I'm stuck at `Carthage` checkout stage. `lipo` is complaining about `same architectures (arm64) and can't be in the same fat output file` while building dependencies.
How did you manage to get around that?

**Comment by MarcoCarnevali** [comment]
@Udumft I actually don't know... I haven't touched Carthage but I used Cocoapods and I can confirm you that with it works good

File: MapboxNavigation/CarPlayManager.swift [hunk 1]
```diff
@@ -244,7 +244,7 @@ public class CarPlayManager: NSObject {
 // MARK: CPApplicationDelegate
 @available(iOS 12.0, *)
 extension CarPlayManager: CPApplicationDelegate {
-    public func application(_ application: UIApplication, didConnectCarInterfaceController interfaceController: CPInterfaceController, to window: CPWindow) {
+    public func application(_ application: UIApplication, didConnectCarInterfaceController interfaceController: CPInterfaceController, to window: CPWindow?) {
```
Comment (Udumft) [inline MapboxNavigation/CarPlayManager.swift:247]:
```suggestion
#if targetEnvironment(macCatalyst)
    public func application(_ application: UIApplication,
                            didConnectCarInterfaceController interfaceController: CPInterfaceController,
                            to window: CPWindow?) {
        handle(application: application,
               didConnectCarInterfaceController: interfaceController,
               to: window)
    }
    #else
    public func application(_ application: UIApplication,
                            didConnectCarInterfaceController interfaceController: CPInterfaceController,
                            to window: CPWindow) {
        handle(application:application,
               didConnectCarInterfaceController: interfaceController,
               to: Optional(window))
    }
    #endif
    
    func handle(application: UIApplication,
                didConnectCarInterfaceController interfaceController: CPInterfaceController,
                to window: CPWindow?) {
                // ...
```

File: MapboxNavigation/CarPlayManager.swift [hunk 2]
```diff
@@ -267,7 +267,7 @@ extension CarPlayManager: CPApplicationDelegate {
         }
     }
 
-    public func application(_ application: UIApplication, didDisconnectCarInterfaceController interfaceController: CPInterfaceController, from window: CPWindow) {
+    public func application(_ application: UIApplication, didDisconnectCarInterfaceController interfaceController: CPInterfaceController, from window: CPWindow?) {
```
Comment (Udumft) [inline MapboxNavigation/CarPlayManager.swift:270]:
```suggestion
#if targetEnvironment(macCatalyst)
    public func application(_ application: UIApplication,
                            didDisconnectCarInterfaceController interfaceController: CPInterfaceController,
                            from window: CPWindow?) {
        handle(application: application,
               didDisconnectCarInterfaceController: interfaceController,
               from: window)
    }
    #else
    public func application(_ application: UIApplication,
                            didDisconnectCarInterfaceController interfaceController: CPInterfaceController,
                            from window: CPWindow) {
        handle(application: application,
               didDisconnectCarInterfaceController: interfaceController,
               from: Optional(window))
    }
    #endif
    
    func handle(application: UIApplication,
                didDisconnectCarInterfaceController interfaceController: CPInterfaceController,
                from window: CPWindow?) {
                // ...
```

**Comment by S2Ler** [comment]
Looks like it was an issue with beta version of the Xcode. Feel free to reopen if it is still an issue.

**Pull request closed without merging**